### PR TITLE
O3-951: Major lag caused by updateConfigExtensionStore

### PR DESCRIPTION
**Pull request by brandones**
## Summary

This improves the performance of loading extensions. The actual fix is in the first commit; I recommend reviewing this PR per-commit. The commits have been organized to facilitate that.

The fix prevents unnecessary calls to `configExtensionStore.setState`, which was causing lots of churn in the React reconciler. This improves the performance of clicking the user menu by about 70%; reducing the response time from about 500ms to about 140ms (when using `run:shell`, which uses a development build of React).

## Screenshots

These are profiles of what happens when you click the user menu.

Before
Screenshot from 2021-12-02 20-26-55

After
Screenshot from 2021-12-02 21-47-41

## Related Issue

https://issues.openmrs.org/browse/O3-951

### CODE REVIEW

**Comment by jonathandick** [comment]
Awesome work @brandones !

File: packages/framework/esm-extensions/src/store.ts [hunk 1]
```diff
@@ -124,5 +125,14 @@ function updateConfigExtensionStore(extensionState: ExtensionStore) {
...
+  if (
+    !isEqual(
+      configExtensionStore.getState().mountedExtensions,
+      configExtensionRecords
+    )
+  ) {
+    configExtensionStore.setState({
+      mountedExtensions: configExtensionRecords,
+    });
+  }
```
Comment (brandones) [inline packages/framework/esm-extensions/src/store.ts:137]: This is the important change for this ticket. Everything else is more or less unrelated.

**Review by denniskigen — APPROVED** [review]
Nicely done, @brandones!

**Review by ibacher — APPROVED** [review]

**Comment by ibacher** [comment]
LGTM. Thanks @brandones!

**Pull request merged**